open_fdd/desktop_bridge/server.py

```
@@ -402,21 +933,44 @@ def data_model_testing_query(body: SparqlQueryBody) -> dict[str, Any]:
-    @app.get("/data-model/ttl")
+    @app.post("/data-model/sparql", tags=["sparql"])
+    def data_model_sparql(body: SparqlTextBody) -> dict[str, Any]:
+        """
+        AFDD-stack style SPARQL endpoint for agent compatibility.
+        Returns list of binding dictionaries under `bindings`.
+        """
+        result = data_model_testing_query(SparqlQueryBody(query=body.query))
+        return {"bindings": result.get("rows", [])}
+
+    @app.post("/data-model/sparql/upload", tags=["sparql"])
+    async def data_model_sparql_upload(file: UploadFile = File(...)) -> dict[str, Any]:
```

Ruff (0.15.12)

[warning] Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)